### PR TITLE
feat: metrics package

### LINKED ISSUE #6: feat: metrics package
## Description of the feature
Migrate metrics package

## Dependencies
Links to other tasks that may depend or be dependent of this one.

### PR DESCRIPTION
**Pull request by wsalles**
closes #6

### CODE REVIEW

**Comment by portellaa** [comment]
please fix the title 😂  it's matrics

**Comment by portellaa** [comment]
@wsalles 

please fix the title of the PR and in the commit, you have matrics instead of metrics
please remove (#6) from the commit title, that's not needed

File: pkg/common/types.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+type UserID string
+
+// RESTRequest defines a struct for a request
+type RESTRequest struct {
+	URL         string
+	Body        interface{}
+	Headers     map[string]string
+	QueryParams map[string]string
+	Timeout     time.Duration
+}
```
Comment (portellaa) [inline pkg/common/types.go:23]: i would move this to another package 🤔 the http one

**Review by portellaa — COMMENTED** [review]
looking good, just a comment.

could we create some documentation on how to use metrics?

File: pkg/common/types.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+func (n Namespace) String() string {
```
Comment (victoriavilasb) [inline pkg/common/types.go:25]: missing doc here

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+// Collect implements the prometheus.Collector interface.
+func (m *Metrics) Collect(ch chan<- prometheus.Metric) {
+	m.scrape()
+	m.runningMetrics.Collect(ch)
+	m.Creation.Collect(ch)
+	m.FailCreation.Collect(ch)
+}
+
+// Scrape gets current running statefulsets.
+func (m *Metrics) scrape() {
```
Comment (victoriavilasb) [inline pkg/metrics/metrics.go:85]: there no need to document private methods

File: pkg/common/types.go (hunk 1 above)
Comment (wsalles) [reply pkg/common/types.go:23]: Done.

File: pkg/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	"github.com/prometheus/client_golang/prometheus"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Metrics includes metrics used in controller
+type Metrics struct {
+	cli              client.Client
```
Comment (portellaa) [inline pkg/metrics/metrics.go:15]: i would prefer to call it client, instead of cli

File: pkg/metrics/metrics.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	name             string
+	runningMetrics   *prometheus.GaugeVec
+	Creation         *prometheus.CounterVec
+	FailCreation     *prometheus.CounterVec
+	CullingCount     *prometheus.CounterVec
+	CullingTimestamp *prometheus.GaugeVec
```
Comment (portellaa) [inline pkg/metrics/metrics.go:21]: 🤔  what if we change those into methods?

Comment (wsalles) [reply pkg/metrics/metrics.go:21]: why?

**Pull request merged**